mivideo/queries.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+
+def dfCourseEventsQuery(lastTime: str) -> str:
+    return f'''
+        SELECT
+        DISTINCT event_hour_utc,
+                 course_id,
+                 MAX(event_time) AS event_time_utc_latest,
+                 COUNT(*) AS event_count
+        FROM
+            (
+            SELECT
+        SAFE_CAST(
+            JSON_EXTRACT_SCALAR(
+                event,
+                '$.object.extensions.kaf:course_id'
+            ) AS INT64
+        )AS course_id,
+            FORMAT_TIMESTAMP(
+                '%F %H',
+                event_time,
+                'UTC'
+            ) AS event_hour_utc,
+                 event_time
+        FROM
+        `udp-umich-prod`.event_store.events
+        WHERE
+            (
+            ed_app = 'https://aakaf.mivideo.it.umich.edu/caliper/info/app/KafEdApp'
+        OR ed_app = 'https://1038472-1.kaf.kaltura.com/caliper/info/app/KafEdApp'
+        )
+        AND event_time > TIMESTAMP('{lastTime}')
```

Comment:
I think `google-cloud-bigquery` has parameter functionality, so we should probably use that for the SQL Injection reasons broached above. See this example from MyLA: https://github.com/tl-its-umich-edu/my-learning-analytics/blob/36a05c27eee83dd702d65666586ebebf0ab4cf0c/dashboard/cron.py#L270-L287

Reply:
✅Fixed.